src/categorical_algebra/CSets.jl

```diff
@@ -643,20 +686,26 @@ that the vertex map is injective but imposes no constraints on the edge map.
 
 To restrict the homomorphism to a given partial assignment, set the keyword
 argument `initial`. For example, to fix the first source vertex to the third
-target vertex in a graph homomorphism, set `initial=(V=Dict(1 => 3),)`.
+target vertex in a graph homomorphism, set `initial=(V=Dict(1 => 3),)`. Use 
+the keyword argument `type_components` to specify nontrivial components on 
+attribute types for a loose homomorphism. These components must be callable:
+either Julia functions on the appropriate types or FinFunction(Dict(...)).
 
 Use the keyword argument `alg` to set the homomorphism-finding algorithm. By
 default, a backtracking search algorithm is used ([`BacktrackingSearch`](@ref)).
+Use the keyword argument error_failures = true to get errors explaining 
+any immediate inconsistencies in specified initial data.
 
 See also: [`homomorphisms`](@ref), [`isomorphism`](@ref).
 """
-homomorphism(X::ACSet, Y::ACSet; alg=BacktrackingSearch(), kw...) =
+homomorphism(X::ACSet, Y::ACSet; alg=BacktrackingSearch(), kw...) = begin 
   homomorphism(X, Y, alg; kw...)
+end
 
 function homomorphism(X::ACSet, Y::ACSet, alg::BacktrackingSearch; kw...)
   result = nothing
   backtracking_search(X, Y; kw...) do α
-    result = α; return true
+    result = isa(α,ACSetTransformation) ? α : nothing; return true
```

Comment:
Why is this check needed? My guess is that this is leftover from a previous design.